dtls/server-dtls13-event.c

```diff
@@ -0,0 +1,588 @@
+/* server-dtls13.c
+ *
+ * Copyright (C) 2006-2022 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL. (formerly known as CyaSSL)
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ *=============================================================================
+ *
+ * Single threaded example of a DTLS 1.3 server for instructional/learning
+ * purposes. This example can handle multiple simultaneous connections by using
+ * the libevent library to handle the event loop. Please note that this example
+ * is not thread safe as access to global objects is not protected.
+ */
+
+#include <wolfssl/options.h>
+#include <stdio.h>                  /* standard in/out procedures */
+#include <stdlib.h>                 /* defines system calls */
+#include <string.h>                 /* necessary for memset */
+#include <netdb.h>
+#include <sys/socket.h>             /* used for all socket calls */
+#include <netinet/in.h>             /* used for sockaddr_in */
+#include <arpa/inet.h>
+#include <wolfssl/ssl.h>
+#include <wolfssl/error-ssl.h>
+#include <errno.h>
+#include <signal.h>
+#include <unistd.h>
+
+/* Requires libevent */
+#include <event2/event.h>
```

Comment:
Would be nice for folks missing libevent for the "make" on this DTLS folder to still function. Is there a way to detect libevent and disable this file / feature?

Reply:
I'll see if its possible for Makefile to detect the presence of a lib. Should be possible by checking the return of `gcc -levent`.

Reply:
The `Makefile` now ignores linking errors when compiling `server-dtls13-event.c`.